### PR TITLE
Doc improvements for walk operator

### PR DESCRIPTION
**Pull request by mpharrigan**
provide an intro for the qubitization walk operator; document select and prepare interfaces #980

### CODE REVIEW

**Comment by fdmalone** [comment]
I did not know it was that easy (and didn't try)

File: qualtran/bloqs/qubitization_walk_operator.ipynb [hunk 1]
```diff
@@ -28,6 +42,78 @@
...
+   "cell_type": "markdown",
+   "id": "774308d5",
+   "metadata": {
+    "cq.autogen": "PrepareOracle.bloq_doc.md"
+   },
+   "source": [
+    "## `PrepareOracle`\n",
+    "Abstract base class that defines the API for a PREPARE Oracle.\n",
+    "\n",
+    "Given a set of coefficients $\\{c_0, c_1, ..., c_{N - 1}\\}$, the PREPARE oracle is used to encode\n",
```
Comment (fdmalone) [inline qualtran/bloqs/qubitization_walk_operator.ipynb:92]: nit: should probably be $\ell$ or $l$ here to match SELECT docs.

File: qualtran/bloqs/qubitization_walk_operator.ipynb [hunk 2]
```diff
@@ -28,6 +42,78 @@
...
+    "the coefficients as amplitudes of a state $|\\Psi\\rangle = \\sum_{i=0}^{N-1} \\sqrt{\\frac{c_{i}}{\\lambda}} |i\\rangle$\n",
+    "where $\\lambda = \\sum_i |c_i|$, using a selection register $|i\\rangle$. In order to prepare such\n",
+    "a state, the PREPARE circuit is also allowed to use a junk register that is entangled with\n",
+    "selection register.\n",
+    "\n",
+    "Thus, the action of a PREPARE circuit on an input state $|0\\rangle$ can be defined as:\n",
+    "\n",
+    "$$\n",
+    "    \\mathrm{PREPARE} |0\\rangle = \\sum_{i=0}^{N-1} \\sqrt{ \\frac{c_{i}}{\\lambda} } |i\\rangle |junk_{i}\\rangle\n",
```
Comment (fdmalone) [inline qualtran/bloqs/qubitization_walk_operator.ipynb:101]: nit: \mathrm{junk}

**Review by fdmalone — APPROVED** [review]
Maybe out of scope for this PR but the latex in QubitizationWalkOperator looks off. I thought this was fixed a while back, but maybe not.

File: qualtran/bloqs/qubitization_walk_operator.ipynb (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/qubitization_walk_operator.ipynb:92]: $l$ refers to any given index not the final one though, right?

Comment (fdmalone) [reply qualtran/bloqs/qubitization_walk_operator.ipynb:92]: Sorry I clicked the wrong line for this comment:  I meant the next line where you use $\sum_i$

**Comment by mpharrigan** [comment]
@fdmalone I accidentally went through the `QubitizationWalkOperator` as well. Can you take a look? There was a mix of \ell and L -- I standardized on L

File: qualtran/bloqs/qubitization_walk_operator.ipynb (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/qubitization_walk_operator.ipynb]: AH yes

File: qualtran/bloqs/select_and_prepare.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def signature(self) -> Signature:
 class PrepareOracle(GateWithRegisters):
     r"""Abstract base class that defines the API for a PREPARE Oracle.
 
-    Given a set of coefficients $\{c_0, c_1, ..., c_{N - 1}\}$, the PREPARE oracle is used to encode
+    Given a set of coefficients $\{c_0, c_1, ..., c_{l - 1}\}$, the PREPARE oracle is used to encode
```
Comment (mpharrigan) [inline qualtran/bloqs/select_and_prepare.py:75]: oops

**Comment by mpharrigan** [comment]
image

File: qualtran/bloqs/select_and_prepare.py (hunk 3 above)
Comment (mpharrigan) [reply qualtran/bloqs/select_and_prepare.py]: fixed

**Comment by fdmalone** [comment]
Beautiful

**Pull request merged**